weaver/cli.py

```diff
@@ -398,13 +435,75 @@ def _parse_inputs(inputs):
                 raise ValueError("Input values parsed as null. Could not properly detect employed schema.")
+            values = convert_input_values_schema(values, schema=PROCESS_SCHEMA_OGC)
         except Exception as exc:
             return OperationResult(False, f"Failed inputs parsing with error: [{exc!s}].", inputs)
         return values
 
+    def _update_files(self, inputs, url=None):
+        # type: (ExecutionInputsMap, Optional[str]) -> Tuple[ExecutionInputsMap, HeadersType]
+        """
+        Replaces local file paths by references uploaded to the :term:`Vault`.
+
+        .. seealso::
+            - Headers dictionary limitation by :mod:`requests`:
+              https://docs.python-requests.org/en/latest/user/quickstart/#response-headers
+            - Headers formatting with multiple values must be provided by comma-separated values
+              (:rfc:`7230#section-3.2.2`).
+            - Multi Vault-Token parsing accomplished by :func:`weaver.vault.utils.parse_vault_token`.
+            - More details about formats and operations related to :term:`Vault` are provided
+              in :ref:`file_vault_token` and :ref:`vault` chapters.
+
+        :param inputs: Input values for submission of :term:`Process` execution.
+        :return: Updated inputs.
+        """
+        auth_tokens = {}  # type: Dict[str, str]
+        update_inputs = dict(inputs)
+        for input_id, input_data in dict(inputs).items():
+            if not isinstance(input_data, list):  # support array of files
+                input_data = [input_data]
+            for data in input_data:
+                if not isinstance(data, dict):
+                    continue
+                file = href = get_any_value(data, default=null, data=False)
+                if not isinstance(href, str):
+                    continue
+                if href.startswith("file://"):
+                    href = href[7:]
+                if "://" not in href and not os.path.isfile(href):
+                    LOGGER.warning(
+                        "Ignoring potential local file reference since it does not exist. "
+                        "Cannot upload to vault: [%s]", file
+                    )
+                    continue
+                fmt = data.get("format", {})
+                ctype = get_field(fmt, "mime_type", search_variations=True)
+                if not ctype:
+                    ext = os.path.splitext(href)[-1]
+                    ctype = get_content_type(ext)
+                fmt = get_format(ctype, default=CONTENT_TYPE_TEXT_PLAIN)
+                res = self.upload(href, content_type=fmt.mime_type, url=url)
+                if res.code != 200:
+                    return res
+                vault_href = res.body["file_href"]
+                vault_id = res.body["file_id"]
+                token = res.body["access_token"]
+                auth_tokens[vault_id] = token
+                LOGGER.info("Converted [%s] -> [%s]", file, vault_href)
```

Comment:
In the context of the CLI it is intentional since the user should know which file they are themselves uploading anyway.  
It would be unintentional in other places (API side). I think I only added debug messages in those cases.